### PR TITLE
some changes to list dataset columns

### PR DESCRIPTION
**Pull request by dcaseykc**
Main changes:
1) Improve HYS column generation
2) Implement analytic_only flag as intended. Analytic ready variables cannot be entirely inferred by whether they have `chi_` as a prefix. Instead, its more of a schema/versioning thing (e.g. stage vs final tables).

### CODE REVIEW

**Comment by dcaseykc** [comment]
To do: Should we add unit tests? How would that work since the underlying data is liable to change?

File: R/utilities.R [hunk 1]
```diff
@@ -821,42 +821,40 @@ list_dataset_columns <- function(dataset, year = 2021, analytic_only = F){
...
     }
-    fps = file.path('//PHDATA01/EPE_Data/HYSdata/hys/2021/v1/', paste0('hys_ar_', year, '.rds'))
-    dat <- data.table::rbindlist(lapply(fps, readRDS), use.names = T, fill = T)
-    var.names <- names(dat)
+
+    var.names = c(var.names.ar, var.names.stg)
+    ar = c(rep(TRUE, length(var.names.ar)), rep(FALSE, length(var.names.stg)))
 
   } else {
     stop(paste0('list_dataset_columns functionality for dataset "', dat, '" not currently available/implemented'))
```
Comment (rwbuie) [inline R/utilities.R:854]: [dat] in this line should be changed to [dataset]

File: R/utilities.R [hunk 2]
```diff
@@ -821,42 +821,40 @@ list_dataset_columns <- function(dataset, year = 2021, analytic_only = F){
   # create a negate function of %in% for readability
   '%!in%' = Negate('%in%')
 
-  if(dataset %!in% list_apde_data()) {
-    dat = match.arg(dataset, list_apde_data())
-    warning(paste0('non-exact database chosent. Attempting best match using "', dat,'" instead of "', dataset, '".'))
-  }
+  dataset = match.arg(dataset, c('birth', 'hys'))
```
Comment (rwbuie) [inline R/utilities.R:824]: If keeping match.arg(), my thinking is the user should receive feedback about matches made if made. Why remove the warning?

Comment (dcaseykc) [reply R/utilities.R:831]: In the most recent commit, I've combined the warning and error into one. Now if the user asks for a dataset that is not implemented, it'll error out and tell them which ones are implemented.

Comment (rwbuie) [reply R/utilities.R:831]: then, should this match.arg operation be removed?

Comment (dcaseykc) [reply R/utilities.R]: Good point. The most recent commit has it removed and instead I add a stopifnot to prevent dataset from being not length 1

**Pull request merged**